src/app/redux/Game/reducers.js

```diff
@@ -0,0 +1,73 @@
+import * as Immutable from 'seamless-immutable';
+
+import { editSquare } from '../../../utils/utils';
+import { lines } from '../../screens/Game/constants';
+
+import { actions } from './actions';
+
+const initialState = {
+  stepNumber: 0,
+  xIsNext: true,
+  winner: '',
+  squareList: [
+    { id: 0, value: '' },
+    { id: 1, value: '' },
+    { id: 2, value: '' },
+    { id: 3, value: '' },
+    { id: 4, value: '' },
+    { id: 5, value: '' },
+    { id: 6, value: '' },
+    { id: 7, value: '' },
+    { id: 8, value: '' }
+  ],
+  history: [
+    {
+      squareList: [
+        { id: 0, value: '' },
+        { id: 1, value: '' },
+        { id: 2, value: '' },
+        { id: 3, value: '' },
+        { id: 4, value: '' },
+        { id: 5, value: '' },
+        { id: 6, value: '' },
+        { id: 7, value: '' },
+        { id: 8, value: '' }
+      ]
+    }
+  ]
+};
+
+function calculateWinner(squares) {
+  for (let i = 0; i < lines.length; i += 1) {
+    const [a, b, c] = lines[i];
+    if (squares[a].value && squares[a].value === squares[b].value && squares[a].value === squares[c].value) {
+      return squares[a].value;
+    }
+  }
+  return null;
+}
+
+function reducer(state = Immutable(initialState), action) {
+  switch (action.type) {
+    case actions.CHANGE_SQUARE_STATUS: {
+      const newList = editSquare(state.squareList, action.payload.idSquare, action.payload.value);
+      if (state.winner || state.squareList[action.payload.idSquare].value) return state;
```

Comment:
if the game is over, or the square already has a value, i think you shouldnt dispatch the action. What do you think?